### PR TITLE
JavaScript Port

### PR DESCRIPTION
**Pull request by RobsonMi**
Created port of original code and simple interface for it

### CODE REVIEW

**Comment by apocolipse** [comment]
This is good but can you make them sliders instead of button incrementers?  Buttons or text entry somewhat defeat the flexibility of this tool

File: js/RotorPirates.html [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        <script src="https://code.jquery.com/jquery-3.2.1.slim.min.js" integrity="sha384-KJ3o2DKtIkvYIK3UENzmM7KCkRr/rE9/Qpg6aAZGJwFDMVNA/GpGFF93hXpG5KkN" crossorigin="anonymous"></script>
+        <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0-beta.3/css/bootstrap.min.css" integrity="sha384-Zug+QiDoJOrZ5t4lssLdxGhVrurbmBWopoEl+M6BdEfwnCJZtKxi1KgxUyJq13dy" crossorigin="anonymous">
+        
+        <link rel="stylesheet" href="style.css">
+        <script type="text/javascript" src="https://www.gstatic.com/charts/loader.js"></script>
+
+        <script src="RotorPirates.js"></script>
+    </head>
+    <body>
+        <img src="http://monsterquads.eu/wp-content/uploads/2017/02/monsterquadsXR-log2.png" style="width:400px;margin-top:30px"/>
```
Comment (apocolipse) [inline js/RotorPirates.html]: Free to use on your own but please remove for here ;)

File: js/RotorPirates.html [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        <h3>Rates Calculator</h3>
```
Comment (apocolipse) [inline js/RotorPirates.html]: RotorPirates! (since it started as python-rates->pyrates->RotorPirates)

File: js/RotorPirates.html [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        <div id="curve_chart" style="width: 900px; height: 500px; margin-top:-70px"></div>
+        <div class="rateBox">
+            <span class="block-title">RaceFlight Rates</span>
+            <div>
+                <span class="input-title">Rate:</span>
+                <span class="input-title">Expo:</span>
+                <span class="input-title">Acro+:</span>
+            </div>
+            <div>
+                <input type="number" name="rfRate" min="0" max="2000" value="400" class="input-field"/>
```
Comment (apocolipse) [inline js/RotorPirates.html]: As mentioned, Slider input (type="range") though I'm no web person so not sure how to finagle that in a pretty fashion.
Number inputs are fine but sliders too!!!!!

File: js/RotorPirates.js [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+
+
+// BF rate calculation function
+var bfcalc = function(rcCommand, rcRate, expo, superRate) {
+    var clamp = function(n, minn, maxn) {
+        return Math.max(Math.min(maxn, n), minn);
+    }
+
+    if(rcRate > 2.0)
+        rcRate = rcRate + (14.55 * (rcRate - 2.0))
```
Comment (apocolipse) [inline js/RotorPirates.js]: This should actually be changed to 14.54 (just double checked in BF code), changing in python as well

File: js/RotorPirates.html (hunk 3 above)
Comment (RobsonMi) [reply js/RotorPirates.html]: I thought about it and to me this seems closer to UI in configurators for these FC's.

File: js/RotorPirates.js (hunk 4 above)
Comment (RobsonMi) [reply js/RotorPirates.js]: Okay. I'll change it.

File: js/RotorPirates.html (hunk 3 above)
Comment (apocolipse) [reply js/RotorPirates.html]: It is (and feel free to keep that in there), but Sliders are where the real value is at :)  its difficult to play with the ranges of these values with increment buttons

Comment (RobsonMi) [reply js/RotorPirates.html]: Will work on it

Comment (apocolipse) [reply js/RotorPirates.html]: Cool let me know :) Once you have it done i'll merge in and host on github pages :D

File: js/RotorPirates.html [hunk 5]
```diff
@@ -9,8 +9,7 @@
         <script src="RotorPirates.js"></script>
     </head>
     <body>
-        <img src="http://monsterquads.eu/wp-content/uploads/2017/02/monsterquadsXR-log2.png" style="width:400px;margin-top:30px"/>
```
Comment (apocolipse) [inline js/RotorPirates.html]: BTW this wasn't loading for me anyway, something wrong with the domain?

**Comment by apocolipse** [comment]
Figured out sliders :) Merging

**Comment by apocolipse** [comment]
@RobsonMi now live at https://apocolipse.github.io/RotorPirates/ :) Thanks sir!

**Comment by RobsonMi** [comment]
No prob. My pleasure ;)

**Pull request merged**